### PR TITLE
Sindhu/Conversion of the below pytests to C++ tests

### PR DESCRIPTION
**Pull request by sindhu-nervana**
1) Math Op test_abs.py
2) Array Op test_fill.py

### CODE REVIEW

File: test/test_math_ops.cpp [hunk 1]
```diff
@@ -389,6 +389,48 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
   opexecuter.RunTest();
 }
 
+TEST(MathOps, Abs1d) {
```
[the remark this answers is not in the document]
Comment (sindhu-nervana) [reply test/test_math_ops.cpp]: I made the changes.

File: test/test_math_ops.cpp [hunk 2]
```diff
@@ -389,6 +389,47 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
...
+  Scope root = Scope::NewRootScope();
+  int dim1 = 1;
+
+  Tensor A(DT_FLOAT, TensorShape({dim1}));
```
Comment (sayantan-nervana) [inline test/test_math_ops.cpp]: By abs1d, we mean a abs test on a rank1 tensor (or a vector).
If my understanding is right this is how you might define a vector of length 5
```
TensorShape({dim1})
```

The fact that there is only 1 number in the braced list means it is rank 1. however that number itself can be non-one.

In this test you have dim1 set to 1. So '1' becomes a bit overloaded (rank is 1 and dim is also 1). Also we are testing a degenerate/trivial vector since it is basically a single number

TL;DR: set dim1 to some integer > 1

File: test/test_array_ops.cpp [hunk 3]
```diff
@@ -157,6 +157,40 @@ TEST(ArrayOps, Unpack) {
...
+  Tensor input_data(DT_FLOAT, TensorShape({}));
+  AssignInputValuesRandom(input_data);
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  // Fill creates a tensor filled with scalar value
+  // 1-D shape of the output tensor
+  auto shape1 =
+      ops::Const(root1, {int32(2), int32(1), int32(3)}, TensorShape({3}));
+  auto R1 = ops::Fill(root1, shape1, input_data);
```
Comment (sayantan-nervana) [inline test/test_array_ops.cpp]: So earlier we ran into the problem of incorrect types and casting. You seem to have figured out the right types to use here ```int32(2)```:

So now you can have a loop over different sizes.
```
vector<vector<i32>> shape_vector= {{2,1,3}, {1,2}, {4,5}};
...
for (auto shape : shape_vector)
{
  auto shape_const = ops::Const(root1, shape, TensorShape({shape.size()});
  ...
}
```

File: test/test_math_ops.cpp (hunk 2 above)
Comment (sindhu-nervana) [reply test/test_math_ops.cpp]: Changed.

File: test/test_array_ops.cpp (hunk 3 above)
Comment (sayantan-nervana) [reply test/test_array_ops.cpp]: As discussed, you are right, we cannot use vectors. Please checkout slack for a workaround

Comment (sindhu-nervana) [reply test/test_array_ops.cpp]: I did try to use the input_sizes vector and loop for different sizes but that did not work as Fill op is not taking pre-defined vectors. So used constant and assigned int32 values to the tensor.

File: test/test_array_ops.cpp [hunk 4]
```diff
@@ -157,6 +157,40 @@ TEST(ArrayOps, Unpack) {
...
+
+  vector<int> static_input_indexes = {0};  // has static input
+
+  for (auto const& input_size : input_sizes) {
+    Scope root = Scope::NewRootScope();
+
+    int input_dim = input_size.size();
+    Tensor shape(DT_INT32, TensorShape({input_dim}));
+    AssignInputIntValues(shape, 5);
+
```
Comment (sayantan-nervana) [inline test/test_array_ops.cpp]: I think this is not doing what we want.
Lets say: ```input_size = {2,3,4}```
Then, ```input_dim = 3```
Then ```shape``` is a int32 tensor of dims ```1x3``` (or just 3, not exactly sure)
Then ```AssignInputIntValues(shape, 5);``` is saying fill up this ```1x3``` tensor names ```shape``` with values starting from 0 and increasing by int steps. So ```shape``` tensor would now be ```{0,1,2}```.

Which means we will create a tensor of size ```0,1,2``` and not ```{2,3,4}``` as originally intended.

Can you please check and confirm this.

**Comment by sayantan-nervana** [comment]
Maybe we add a new utility? (unless it already exists?)
```
template <typeclass T>
void AssignInputValuesFromVector(Tensor& A, vector<T> x) {
  auto A_flat = A.flat<float>();
  auto A_flat_data = A_flat.data();
  assert A_flat.size()==x.size()
  for (int i = 0; i < A_flat.size(); i++) {
    A_flat_data[i] = x[i];
  }
}
```

File: test/test_utilities.h [hunk 5]
```diff
@@ -41,6 +41,16 @@ void AssignInputValuesRandom(Tensor& A);
...
+void AssignInputValuesFromVector(Tensor& A, vector<T> x) {
+  auto A_flat = A.flat<T>();
+  auto A_flat_data = A_flat.data();
+  assert(A_flat.size() == x.size());
+  for (int i = 0; i < A_flat.size(); i++) {
+    A_flat_data[i] = x[i];
+  }
+}
+
 template <class T>
```
Comment (sayantan-nervana) [inline test/test_utilities.h]: Could you please remive the function declarations, if they are not being used anywhere
AssertTensorEqualsFloat
AssertTensorEqualsIntInt32
